File: Cpp/347.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> ans;
        unordered_map<int,int> m;
        for(int i = 0; i < nums.size(); i++){
            m[nums[i]]++;
        }
        vector<vector<int>> x(nums.size()+1);
        for(auto it = m.begin(); it != m.end(); it++){
            x[it->second].push_back(it->first);
        }
        int ctr=0;
        
        for(int i = x.size()-1; i >= 0; i--){
            if(ctr >= k)break;
            for(int j = 0; j < x[i].size(); j++){
                ans.push_back(x[i][j]);
                ctr++;
            }
        }
         return ans;
        
    }
};

File: Python/347.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        fd = {}
        for num in nums:
            if num in fd.keys():
                fd[num] = fd[num]+1
            else:
                fd[num] = 1
        arr = ['0']* len(nums)
        for key, v in fd.items():
            if arr[v-1] == '0':
                arr[v-1] = [key]
            else:
                arr[v-1].append(key)
        sol = []
        for i in reversed(range(len(nums))):
            if len(sol) == k:
                break
            if arr[i] != '0':
                sol = sol + arr[i]
        return sol
